src/System.Text.Json/src/System/Text/Json/Node/JsonBoolean.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json
+{
+    /// <summary>
+    ///   Represents a boolean JSON value.
+    /// </summary>
+    public class JsonBoolean : JsonNode, IEquatable<JsonBoolean>
+    {
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonBoolean"/> class representing the value <see langword="false"/>.
+        /// </summary>
+        public JsonBoolean() => Value = false;
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonBoolean"/> class representing a specified value.
+        /// </summary>
+        public JsonBoolean(bool value) => Value = value;
+
+        /// <summary>
+        ///   Gets or sets the boolean value represented by the instance.
+        /// </summary>
+        public bool Value { get; set; }
+
+        /// <summary>
+        ///   Converts a <see cref="bool"/> to a JSON boolean.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonBoolean(bool value) => new JsonBoolean(value);
+
+        /// <summary>
+        ///   Compares <paramref name="obj"/> to the value of this instance. 
+        /// </summary>
+        /// <param name="obj">The object to compare against.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the boolean value of this instance matches <paramref name="obj"/>,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public override bool Equals(object obj) => obj is JsonBoolean jsonBoolean && Value == jsonBoolean.Value;
```

Comment:
should this be instead be something like:
```c#
JsonBoolean objAsJsonBoolean = obj as JsonBoolean;
if (objAsJsonBoolean  != null)
{
  return Equals(objAsJsonBoolean);
}
return false;
```

Reply:
I can change it to use equals.

Reply:
> should this be instead be something like

Aside from the use of Equals (which has a redundant check), I don't see a difference between the two.

The `obj is JsonBoolean` check does the null check already. If obj is null, `obj is JsonBoolean` is false.